openpype/hosts/resolve/api/plugin.py

```diff
@@ -418,12 +418,11 @@ def load(self):
         source_in = int(_clip_property("Start"))
         source_out = int(_clip_property("End"))
 
-        if _clip_property("Type") == "Video":
-            source_in += handle_start
-            source_out -= handle_end
-
         # include handles
-        if not self.with_handles:
+        if (
+            not self.with_handles
+            or _clip_property("Type") == "Video"
```

Comment:
Why would a video type **require** handles to be applied?

Reply:
Yeah, that is a very good question. I am not sure myself to be honest. It was there historically and I have already forgotten the reason. In fact, I would also remove the condition. Video or sequence, either it is with handles or not, there should not be a difference in how we are treating it.

Reply:
check this commit if you agree with the solution. fbc24f565415cc6cbb88547c9b333bf41a52a3e0

Now the db frame duration is compared with representation frames availability (the same on sequence or video). This is for case that representation was published with tag `no_handles`. Function is first checking versionData frame start/end and of keys are not available then it looks to the asset data.

Reply:
I disagree with that actually - the asset range is not a replacement for whatever the frame range a published representation would be. I'd say, if no frame range is known that we either get the actual clip's length somehow (does Resolve API allow us to get the length of a clip? I mean the Media Pool can show the length of a clip - I'd assume we could query that somehow?

For example:
```python
media_pool_item.GetClipProperty("Duration")
# 00:00:02:02
media_pool_item.GetClipProperty("End TC")
# 00:00:02:02
media_pool_item.GetClipProperty("Start TC")
# 00:00:00:00
media_pool_item.GetClipProperty("Type")
# Video + Audio
media_pool_item.GetClipProperty("Frames")
# 52
media_pool_item.GetClipProperty("FPS")
# 25
``` 

Why couldn't we use any of that clip information to know the **actual** clip length?

Reply:
I'd say that on load we'd only ever apply the handles of the version or representation then, yes.
If someone wanted to instead apply the handles of the `asset` that should then be an explicit toggle somewhere - I wouldn't expect that to be the default behavior on load.

Reply:
We are trying to obtain frame data from version first and then as fallback we are getting them from the asset. The only reason is that version data are not always having them (although if version is not having them it would be really strange).

Reply:
And what about the case where handles are defined on version data, but representation files are not integrated  handles? 

Anyway, I agree that we should remove the asset data fallback.

Reply:
> And what about the case where handles are defined on version data, but representation files are not integrated handles?

In that case - again, it's an error on the ingestion. If the representation differs from the version then the representation data should contain the differed data. That's an error on the publish itself, not something we need to 'fix' on load.

The frame range should thus be:

- frame range + handles of representation
- if not found, frame range + handles of version

Reply:
>In that case - again, it's an error on the ingestion. If the representation differs from the version then the representation data should contain the differed data. That's an error on the publish itself, not something we need to 'fix' on load.

probably just a difference in opinions, but I don't think it's necessarily an error on ingest. We have plenty of production where .exr representation would have handles, but .mov would not. That way it was easy to quickly assemble the reviewables and see shot hookups in ftrack for instance, by allowing this `no-handles` representation. For episodic animation, it was an absolute must.

Reply:
hey @BigRoy are we ok to merge this PR.

Reply:
> probably just a difference in opinions, but I don't think it's necessarily an error on ingest. We have plenty of production where .exr representation would have handles, but .mov would not.

Doesn't that just mean that we're lacking `frameStart`, `frameEnd`, `handleStart` and `handleEnd` data per representation for those cases?

> That way it was easy to quickly assemble the reviewables and see shot hookups in ftrack for instance, by allowing this no-handles representation. For episodic animation, it was an absolute must.

Totally agree. That's definitely useful. But still, I'd argue that the database would be capable of what file it's providing me instead of somehow saying "just use whatever the asset is set to".

Reply:
This is not guessing of mis-published data so this conversation is misleading. Just to wrap this up before I merge it. Data are published correctly without handles since Extract review output preset is supporting tag `no handles` - this is then supporting to define two representations to be created under one version which is having handles - one representation with hadles and other without handles.

This loading solution is supporting loading both of those representation under one version.